sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/ExternalCatalog.scala

```diff
@@ -184,4 +184,9 @@ abstract class ExternalCatalog {
 
   def listFunctions(db: String, pattern: String): Seq[String]
 
+  // --------------------------------------------------------------------------
+  // Resources
+  // --------------------------------------------------------------------------
+
+  def addJar(path: String): Unit
```

Comment:
Do we have to mention hive here? I'd like to add some documents here to describe the semantic, which can explain why `InMemoryCatalog` can do nothing but `HiveExternalCatalog` need some extra logic